src/api/questions.js

```diff
@@ -298,41 +298,52 @@ router.delete(
 // Mark the question as answered
 router.post(
   '/:questionId/answered',
-  [
-    requireCourseStaffForQueueForQuestion,
-    requireQuestion,
-    check('preparedness').isIn(['bad', 'average', 'good']),
-    check('comments')
-      .optional({ nullable: true })
-      .trim(),
-    failIfErrors,
-  ],
+  [requireCourseStaffForQueueForQuestion, requireQuestion, failIfErrors],
   safeAsync(async (req, res, _next) => {
-    const data = matchedData(req)
-
-    // Temporary, easy fix to avoid having to rename enums
-    // TODO Fix this garbage
-    let mappedPreparedness = data.preparedness
-    switch (data.preparedness) {
-      case 'bad':
-        mappedPreparedness = 'not'
-        break
-      case 'good':
-        mappedPreparedness = 'well'
-        break
-      default:
-        break
-    }
+    const { shouldCheckFeedback } = req.body
+    if (shouldCheckFeedback) {
+      const { feedback } = req.body
+      if (
+        feedback.preparedness !== 'bad' &&
+        feedback.preparedness !== 'good' &&
+        feedback.preparedness !== 'average'
+      ) {
+        res.status(422).send({ error: 'malformed preparedness' })
+      } else {
+        let mappedPreparedness = feedback.preparedness
+        switch (feedback.preparedness) {
+          case 'bad':
+            mappedPreparedness = 'not'
+            break
+          case 'good':
+            mappedPreparedness = 'well'
+            break
+          default:
+            break
+        }
 
-    const { question } = res.locals
-    question.answerFinishTime = new Date()
-    question.dequeueTime = new Date()
-    question.preparedness = mappedPreparedness
-    question.comments = data.comments
-    question.answeredById = res.locals.userAuthn.id
+        const { question } = res.locals
+        question.answerFinishTime = new Date()
+        question.dequeueTime = new Date()
+        question.preparedness = mappedPreparedness
+        question.comments =
+          feedback.comments == null
+            ? feedback.comments
+            : feedback.comments.trim()
+        question.answeredById = res.locals.userAuthn.id
+
+        const updatedQuestion = await question.save()
+        res.status(200).send(updatedQuestion)
+      }
+    } else {
+      const { question } = res.locals
```

Comment:
There's a fair amount of duplicated code between these two branches - you could probably refactor to something like:

```js
const { question } = res.locals
if (shouldCheckFeedback) {
  question.comments = 'here be comments'
  // actual code here...
}
question.answerFinishTime = new Date()
const updatedQuestion = await question.save()
res.status(200).send(updatedQuestion)
```